rt/transports/jms/src/test/java/org/apache/cxf/transport/jms/util/MessageListenerTest.java

```diff
@@ -174,33 +183,21 @@ private void testTransactionalBehaviour(Connection connection, Queue dest) throw
     }
 
     private static Connection createConnection(String name) throws JMSException {
-        ActiveMQConnectionFactory cf = new ActiveMQConnectionFactory("vm://" + name
-                                                                     + "?broker.persistent=false");
-        cf.setRedeliveryPolicy(redeliveryPolicy());
+        ActiveMQConnectionFactory cf = new ActiveMQConnectionFactory("vm://" + name);
         Connection connection = cf.createConnection();
         connection.start();
         return connection;
     }
 
     private static Connection createXAConnection(String name, TransactionManager tm) throws JMSException {
-        ActiveMQXAConnectionFactory cf = new ActiveMQXAConnectionFactory("vm://" + name
-                                                                         + "?broker.persistent=false&jms.xaAckMode=1");
-        cf.setRedeliveryPolicy(redeliveryPolicy());
-        XaPooledConnectionFactory cfp = new XaPooledConnectionFactory(cf);
-        cfp.setTransactionManager(tm);
-        cfp.setConnectionFactory(cf);
-        Connection connection = cfp.createConnection();
+        ActiveMQXAConnectionFactory cf = new ActiveMQXAConnectionFactory("vm://" + name);
+        ConnectionFactory cf1 = new ConnectionFactoryProxy(cf, new TransactionHelperImpl(tm));
```

Comment:
@ehsavoie I found a simpler way to integrate with JTA using `ConnectionFactoryProxy` and `TransactionHelperImpl`, the tests are passing right now (except `testConnectionProblemXA`, it has different cause)

Reply:
@ehsavoie awesome! Could you submit a PR against [this branch](https://github.com/reta/cxf/tree/artemis.attempt.2) (it it works for you), thank you!